MapboxNavigation/PollyVoiceController.swift

```diff
@@ -189,7 +189,7 @@ public class PollyVoiceController: RouteVoiceController {
         }
         
         guard let url = awsTask.result else {
-            callSuperSpeak(fallbackText, error: NSError.withCustom(message: "No AWS Task"))
+            callSuperSpeak(fallbackText, error: NSError(localizedFailureReason: "Unable to read instruction aloud", detailedFailureReason: "No AWS Task"))
```

Comment:
What I meant in https://github.com/mapbox/mapbox-navigation-ios/pull/800#discussion_r149533375 is that this error would contain:

* `NSLocalizedFailureReasonErrorKey`: `Unable to say “In 5 miles, turn left onto Main Street”`
* `SpokenInstructionErrorCodeKey`: SpokenInstructionErrorCode.noAwsTask

The application can then do something different by switching on SpokenInstructionErrorCodeKey rather than doing string comparisons.